content_scripts/mode_find.coffee

```diff
@@ -0,0 +1,44 @@
+# NOTE(smblott).  Ultimately, all of the FindMode-related code should be moved to this file.
+
+# When we use find mode, the selection/focus can end up in a focusable/editable element.  In this situation,
+# PostFindMode handles two special cases:
+#   1. Be an InsertModeBlocker.  This prevents keyboard events from dropping us unintentionaly into insert
+#   mode.  Here, this is achieved by inheriting from InsertModeBlocker.
+#   2. If the very-next keystroke is Escape, then drop immediately into insert mode.
+#
+class PostFindMode extends InsertModeBlocker
```

Comment:
Cool. Tangentially related, but could/how would #556 be implemented in this new and wonderful world of modes?